### PR TITLE
Initial smoke test idea

### PR DESCRIPTION
**Pull request by csibbitt**
* Runs collectd w/ amqp1 plugin inside the saf-telemetry namespace in the cluster
* Collects CPU metrics and pushes it to the bus
* Checks for those CPU metrics (okay, ANY cpu data at the moment) in prometheus
* This validates the qdr router, smart gateway, and prometheus typically in under 10s

### CODE REVIEW

**Comment by csibbitt** [comment]
# Testing

## Negative Testing

I performed two types of negative testing

### Broken collectd->qdr

1. I editted minimal-collectd.conf to have an invalid hostname for the qdr
1. Ran ./smoketest.sh and examined the results:
```
$ oc get po -l name=saf-smoketest
NAME                  READY     STATUS    RESTARTS   AGE
saf-smoketest-kqdbz   0/1       Error     2          31s

$ oc logs saf-smoketest-kqdbz 
grep: /tmp/collectd_output: No such file or directory
Sleeping for 3 seconds waiting for collectd to enter read-loop
plugin_load: plugin "cpu" successfully loaded.
plugin_load: plugin "amqp1" successfully loaded.
Initialization complete, entering read-loop.
amqp1 plugin: PN_TRANSPORT_CLOSED: proton:io: Name or service not known - connect to  ZZqdr-white.sa-telemetry.svc.cluster.local:5672
amqp1 plugin: PN_TRANSPORT_CLOSED: proton:io: Name or service not known - connect to  ZZqdr-white.sa-telemetry.svc.cluster.local:5672
amqp1 plugin: PN_TRANSPORT_CLOSED: proton:io: Name or service not known - connect to  ZZqdr-white.sa-telemetry.svc.cluster.local:5672
Initialization complete, entering read-loop.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   129  100    63  100    66  14106  14778 --:--:-- --:--:-- --:--:-- 16500
amqp1 plugin: PN_TRANSPORT_CLOSED: proton:io: Name or service not known - connect to  ZZqdr-white.sa-telemetry.svc.cluster.local:5672
```

### Expected metrics not appearing in prometheus
1. I reset my prometheus like so: `k delete po prometheus-white-0`
1. I editted minimal-collectd.conf to have no cpu plugin
1. Ran ./smoketest.sh and examined the results:
```
$ oc get po -l name=saf-smoketest
NAME                  READY     STATUS             RESTARTS   AGE
saf-smoketest-trrdh   0/1       Error              2          49s

$ oc logs -f saf-smoketest-trrdh 
plugin_load: plugin "amqp1" successfully loaded.
Sleeping for 3 seconds waiting for collectd to enter read-loop
Initialization complete, entering read-loop.
Initialization complete, entering read-loop.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   129  100    63  100    66    489    512 --:--:-- --:--:-- --:--:--   515
```

## Positive Testing
1. Removed all config edits `git checkout minimal-collectd.conf`
1. Ran ./smoketest.sh and examined the results:
```
$ oc get job
NAME            DESIRED   SUCCESSFUL   AGE
saf-smoketest   1         1            6s

$ oc get po -l name=saf-smoketest
NAME                  READY     STATUS      RESTARTS   AGE
saf-smoketest-f4644   0/1       Completed   0          4m

```
See README.md for more details on how positive test results look.

**Comment by csibbitt** [comment]
[Lightweight smoke test for SAF](https://trello.com/c/mLbDFqiq/581-lightweight-smoke-test-for-saf)

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# SAF Testing notes
+
+Here are some artifacts to assist with testing the SAF after it is deployed.
+
+Currently this is just a "smoke test" that runs internal to the OCP cluster. It delivers collectd data to the SAF amqp node and verifies that it can be seen in prometheus. This is intended to be usable for developers and TravisCI to validate our builds before merging changes to this repo.
```
Comment (leifmadsen) [inline tests/README.md]: A bit nitty, but can you wrap the text to 80 chars?

In vim, just highlight in visual mode and press `gq`. At least that's how my vim is setup, not sure if that's a stock option :)

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Usage
+
+1. Have `oc` pointing at your sa-telemetry project and run `./smoketest.sh`
+1. Run `oc get jobs` and check the result of the saf-smoketest job
+1. (If necesarry) Check the logs of the saf-smoketest pod
```
Comment (leifmadsen) [inline tests/README.md]: necessary

**Comment by leifmadsen** [comment]
Was this ran against the `minishift` install method? I went through this to test, and for some reason I can't make it work.

All I get in the logs is the following:

```
﻿Sleeping for 3 seconds waiting for collectd to enter read-loop
grep: /tmp/collectd_output: No such file or directory
plugin_load: plugin "cpu" successfully loaded.
plugin_load: plugin "amqp1" successfully loaded.
Initialization complete, entering read-loop.
Initialization complete, entering read-loop.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   129  100    63  100    66   9275   9717 --:--:-- --:--:-- --:--:-- 11000
```

Then it just seems to bail out. What whatever reason it seems like it's running into an issue with writing logs out to the `/tmp/` file?

**Comment by csibbitt** [comment]
Yes, it was run against the minishift installation. The initial error of `grep: /tmp/collectd_output: No such file or directory` comes from https://github.com/redhat-service-assurance/telemetry-framework/pull/34/files#diff-d6660f5506a8b7402462a9a4aa125bd3R12 checking before the file appears, which is why it retries. Totally expected (perhaps I should suppress the message)?

Were all your pods "Ready" before you ran the test? That's the only thing that really comes to mind; either that or the egrep line is somehow fragile.

**Comment by csibbitt** [comment]
I'm going to go ahead and merge this; it's not dangerous and if we find the egrep needs adjustment that can come in a 1-liner.

**Pull request merged**